### PR TITLE
Update git_intro.qmd - File staging and git commit

### PR DESCRIPTION
**Pull request by MartinPippel**
- add block for file staging and commits

### CODE REVIEW

File: git/git_intro.qmd [hunk 1]
```diff
@@ -23,6 +23,29 @@ git clone https://github.com/user/repo
 
 ## File staging and git commit
 
+Staging in Git involves adding new, modified, or deleted files to a staging area before committing them. This allows for flexibility in choosing the files to commit.
+
+1. Check status via `git status` You'll see what branch you are on (which for new repos will be master) and status of files (untracked, modified, or deleted).
```
Comment (gbdias) [inline git/git_intro.qmd:28]: new repos get "main" instead of "master"

**Review by gbdias — COMMENTED** [review]
Just one comment added, but everything else looks correct to me.

File: git/git_intro.qmd (hunk 1 above)
Comment (MartinPippel) [reply git/git_intro.qmd:28]: I got it from here: 
https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell#:~:text=A%20branch%20in%20Git%20is,branch%20pointer%20moves%20forward%20automatically.
and here:
https://www.reddit.com/r/git/comments/14gcbx3/can_i_change_the_default_branch_from_master_to/

But I can also remove that part, as its probably freely configurable.

Comment (gbdias) [reply git/git_intro.qmd]: Oh interesting, it must be a version thing then. On my mac (git v 2.39.3) when I init a repo it gets "main", but on NAC cluster (git v2.17.1) it gets "master". It doesn't really matter though as you said this is configurable.

**Review by gbdias — APPROVED** [review]

File: git/git_intro.qmd [hunk 2]
```diff
@@ -23,6 +23,29 @@ git clone https://github.com/user/repo
...
+ - Stage a folder: `git add myfolder`
+3. Check status again: `git status` You should see there are changes ready to be committed.
+4. Unstage a File
+  - If you accidental stage something, use the following command to unstage it: `git reset HEAD example.html`
+5. Deleting Files
+  - If you delete files they will appear in git status as deleted, and you must use git add to stage them. Another way to do this is using git rm command, which both deletes a file and stages it all with one command:
+  - `git rm example.html` to remove a file (and stage it)
+  - `git rm -r myfolder` to remove a folder (and stage it)
+6. Commit Files 
+  -  `git commit -m "Message that describes what this change does"`
```
Comment (mahesh-panchal) [inline git/git_intro.qmd:41]: Suggestion: Also add in a point on `git commit --amend` here to fix commit messages, or add new staged files to the previous commit.

**Review by mahesh-panchal — APPROVED** [review]
Thank you. Looks great.
One suggestion. I'll await your feedback before merging.

**Pull request merged**